### ISSUE TITLE
Usage report & minor annoyance

### ISSUE THREAD

**Issue by ryanprior** [bepvte/gtk-variant.el#2]
Hi, I found this package via Sacha Chua's blog & can report it looks great on elementary OS. Unfortunately, when I start Emacs the frame is light until the end of my init and then flips to dark. I've tried a few strategies to try to use a dark frame earlier, but no luck so far. Do you know of any workaround?

*** Software versions
I'm using GNU Emacs 26.3 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.14)
with Spacemacs (develop branch)

*** Things I've tried

Adding the suggested usage hooks to my `user-init`, which is run before layer configuration:
```elisp
(defun dotspacemacs/user-init ()
  ;; [...]
  (add-hook 'window-setup-hook #'gtk-variant-set-frame)
  (add-hook 'after-make-frame-functions #'gtk-variant-set-frame))
```

Adding an imperative function call during `user-init` or `user-config`:

```elisp
(gtk-variant-set-frame (selected-frame) 'dark)
```

Thanks for creating this! If you have any ideas for avoiding the flash of light frame during startup, I'd  be interested to hear. If there's nothing that can be done, feel free to close this. :smile:

**Comment by bepvte** [comment]
You probably need to put it as early as possible in the init process. I don't use spacemacs so I dont exactly know how that would be done, but I suggest asking on their repo. I'm fairly sure they have some hooks or method for that. 

If their system doesnt work, and the spacemacs package loading time causes a flash of light frame anyway, here are some potential solutions:

* I might try putting the two hooks in spacemacs' built-in `init.el` directly and using `load` instead of `require` to avoid the package system. This is probably not advised by the spacemacs guys.
* Using `emacsclient` instead of `emacs`. Emacsclient connects to a version of emacs that runs in the background, which gets rid of a lot of the startup time and runs the frame setting instantly, as the package is loaded by the emacs "server". I added this to my .zshrc (you might use .bashrc) to make it easier to use: `function ec() { emacsclient -c --alternate-editor= $@ }`. I also copied `/usr/share/applications/emacs.desktop` to `~/.local/share/applications/emacs.desktop` and set `Exec=emacsclient -c -a "" %f` in there.
